### PR TITLE
Create recipe to add lint task with ESLint

### PR DESCRIPTION
**Pull request by felixzapata**
[Polymer uses now ESLint](https://github.com/Polymer/polymer/pull/3382) so I´ve created this recipe to use the tool.

The idea of this pull-request is from the comment of @chuckh in the [pull-request 707](https://github.com/PolymerElements/polymer-starter-kit/pull/707#issuecomment-186224704)

### CODE REVIEW

**Comment by chuckh** [comment]
Thanks @felixzapata!  LGTM.  :+1: :tada:

File: docs/add-eslint-support.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    "browser": true
+  },
+  "plugins": [
+    "html"
+  ],
+  "globals": {
+    "CustomElements": false,
+    "HTMLImports": false,
+    "Polymer": false,
+    "WeakMap": false
```
Comment (abdonrd) [inline docs/add-eslint-support.md]: I think that we don't need all the globals for PSK.

File: docs/add-eslint-support.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
++ var eslint = require('gulp-eslint');
+
+...
+
++ // Lint JavaScript
++ gulp.task('lint', function() {
++   return gulp.src([
++       'app/scripts/**/*.js',
++       'app/elements/**/*.js',
```
Comment (abdonrd) [inline docs/add-eslint-support.md]: In the elements there aren't `.js` files, right?

File: docs/add-eslint-support.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
++       'app/elements/**/*.html',
++       'gulpfile.js'
++     ])
++     .pipe(reload({
++       stream: true,
++       once: true
++     }))
+
++   // JSCS has not yet a extract option
```
Comment (abdonrd) [inline docs/add-eslint-support.md]: Why this comment? :sweat_smile:

**Comment by abdonrd** [comment]
Great work @felixzapata! :tada:

File: docs/add-eslint-support.md (hunk 1 above)
Comment (felixzapata) [inline docs/add-eslint-support.md]: Maybe only Polymer value

Comment (abdonrd) [inline docs/add-eslint-support.md]: Yes, I think so.

File: docs/add-eslint-support.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+## Integrating the lint task
+
+Make sure the `lint` gulp task is triggered by the common build tasks:
+
+ - In the gulp `serve` task, make sure `lint` is triggered initially and on HTML and JS files changes:
+
+```patch
+-gulp.task('serve', ['styles', 'elements', 'images'], function () {
++gulp.task('serve', ['lint', 'styles', 'elements', 'images'], function () {
```
Comment (robdodson) [inline docs/add-eslint-support.md]: I don't think we currently have `images` in the serve task

**Comment by robdodson** [comment]
This recipe looks great but it produces a ton of linter errors:

```
app/elements/my-list/my-list.html
  43:25  error  Unexpected trailing comma  comma-dangle

app/elements/routing.html
  21:7   error  'page' is not defined  no-undef
  21:17  error  'app' is not defined   no-undef
  26:7   error  'app' is not defined   no-undef
  31:7   error  'app' is not defined   no-undef
  36:5   error  'page' is not defined  no-undef
  40:5   error  'page' is not defined  no-undef
  41:7   error  'app' is not defined   no-undef
  44:5   error  'page' is not defined  no-undef
  44:10  error  'app' is not defined   no-undef
  45:7   error  'app' is not defined   no-undef
  48:5   error  'page' is not defined  no-undef
  49:7   error  'app' is not defined   no-undef
  52:5   error  'page' is not defined  no-undef
  53:7   error  'app' is not defined   no-undef
  54:7   error  'app' is not defined   no-undef
  57:5   error  'page' is not defined  no-undef
  58:7   error  'app' is not defined   no-undef
  62:5   error  'page' is not defined  no-undef
  63:7   error  'app' is not defined   no-undef
  64:7   error  'app' is not defined   no-undef
  65:7   error  'page' is not defined  no-undef
  65:21  error  'app' is not defined   no-undef
  69:5   error  'page' is not defined  no-undef

gulpfile.js
   13:12  error  'require' is not defined            no-undef
   14:9   error  'require' is not defined            no-undef
   15:11  error  'require' is not defined            no-undef
   16:19  error  'require' is not defined            no-undef
   17:19  error  'require' is not defined            no-undef
   19:13  error  'require' is not defined            no-undef
   20:12  error  'require' is not defined            no-undef
   21:10  error  'require' is not defined            no-undef
   22:12  error  'require' is not defined            no-undef
   23:26  error  'require' is not defined            no-undef
   24:19  error  'require' is not defined            no-undef
   25:14  error  'require' is not defined            no-undef
   26:19  error  'require' is not defined            no-undef
   27:5   error  'eslint' is defined but never used  no-unused-vars
   27:14  error  'require' is not defined            no-undef
  116:22  error  '__dirname' is not defined          no-undef
  204:48  error  '__dirname' is not defined          no-undef
  312:16  error  'process' is not defined            no-undef
  325:3   error  'require' is not defined            no-undef
  326:15  error  Empty block statement               no-empty
```

I think most of these are missing globals? I'm not super familiar with ESLint but maybe we can fix these in the eslintrc?

File: docs/add-eslint-support.md (hunk 4 above)
Comment (felixzapata) [inline docs/add-eslint-support.md]: hi @robdodson, i've removed the 'images' subtask.

**Comment by felixzapata** [comment]
hi @robdodson, I've added the global variables and this is the new result (only minor bugs):

```
app/elements/my-list/my-list.html
  43:25  error  Unexpected trailing comma  comma-dangle

gulpfile.js
  325:15  error  Empty block statement  no-empty

✖ 2 problems (2 errors, 0 warnings)

```

File: docs/add-eslint-support.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# Add ESLint support
+
+This recipe helps you to create a task to use [ESLint](http://eslint.org/) tool. 
+
+
+## Create .eslintrc file in the root folder
```
Comment (abdonrd) [inline docs/add-eslint-support.md]: `.eslintrc.json` :)

Comment (felixzapata) [inline docs/add-eslint-support.md]: done. It is curious. It works in both ways.

Comment (abdonrd) [inline docs/add-eslint-support.md]: Yes, depends of the ESLint version. From version 2.0.0 should be with extension.

**Comment by robdodson** [comment]
Great! If it isn't too much trouble, would you mind fixing those issues and adding that to this PR as well?

**Comment by robdodson** [comment]
I'll figure out why travis is failing

**Comment by robdodson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by robdodson** [comment]
I think the tests are failing for unrelated reasons. I'm going to go ahead and merge this and we can have a separate issue to fix the tests. Thanks for all your work on this @felixzapata!

**Comment by felixzapata** [comment]
@robdodson you're welcome ;)

**Pull request merged**